src/components/SignInForm/SignInForm.jsx

```diff
@@ -0,0 +1,168 @@
+import { useContext, useState } from "react";
+import styles from "./SignInForm.module.scss";
+import { AuthContext } from "../../containers/AuthContext";
+
+const SignInForm = () => {
+  const authContext = useContext(AuthContext);
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const [errors, setErrors] = useState({});
+
+  const handleEyeButtonClick = (e) => {
+    e.preventDefault();
+    setIsPasswordVisible(!isPasswordVisible);
+  };
+
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+    setErrors((prevErrors) => ({ ...prevErrors, email: null }));
+  };
+
+  const handlePasswordChange = (e) => {
+    setPassword(e.target.value);
+    setErrors((prevErrors) => ({ ...prevErrors, password: null }));
+  };
+
+  const validateEmail = (email) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (!emailRegex.test(email)) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        email: { valid: false, message: "Не правильна адреса" },
+      }));
+      return false;
+    }
+    setErrors((prevErrors) => ({
+      ...prevErrors,
+      email: { valid: true },
+    }));
+    return true;
+  };
+
+  const validatePassword = (password) => {
+    if (password.length < 8) {
```

Comment:
as per my knowledge, we should also validate password for numbers.
please check example in SCRUM-37,

Reply:
ok, I just thought that in this case, since this is a login form, the password already exists on the backend (db) and thorough validation will take place on the backend

Reply:
Done

Reply:
you are actually right :) 
that's a reasonable point. 

but then, same may apply to email, right? 

however, the reason why we still might want to validate both email and password here is: reducing api calls from client to server. if password isn't valid - we don't even want to allow client make api call. coz, server then will have to do some work to check and validate that input, but server is busy with tens or hundreds other users trying to login at the same time. so we don't want to slow the server down :)

Reply:
I agree! Thanks for the explanation